File: FTP Client for WinForms using Ultimate FTP library/[C#]-FTP Client for WinForms using Ultimate FTP library/C#/NewNamePrompt.cs
using System;
using System.Windows.Forms;

namespace ClientSample
{
    public partial class NewNamePrompt : Form
    {
        public NewNamePrompt(string oldName)
        {
            InitializeComponent();

            txtNewName.Text = oldName;
        }

        /// <summary>
        /// Gets the desired new name.
        /// </summary>
        public string NewName
        {
            get
            {
                return txtNewName.Text;
            }
        }

        /// <summary>
        /// Handles the OK button's Click event.
        /// </summary>
        /// <param name="sender">The button object.</param>
        /// <param name="e">The event arguments.</param>
        private void btnOk_Click(object sender, EventArgs e)
        {
            // Check for invalid characters.
            if (NewName.Contains("/") || NewName.Contains("\\"))
            {
                MessageBox.Show("Character '/' or '\\' should not be entered", "New Name Prompt", MessageBoxButtons.OK, MessageBoxIcon.Stop);
                return;
            }

            this.DialogResult = DialogResult.OK;
        }
    }
}

File: FTP Client for WinForms using Ultimate FTP library/[VB]-FTP Client for WinForms using Ultimate FTP library/VB/NewNamePrompt.vb
Namespace ClientSample
	Partial Public Class NewNamePrompt
		Inherits Form
		Public Sub New(ByVal oldName As String)
			InitializeComponent()

			txtNewName.Text = oldName
		End Sub

		''' <summary>
		''' Gets the desired new name.
		''' </summary>
		Public ReadOnly Property NewName() As String
			Get
				Return txtNewName.Text
			End Get
		End Property

		''' <summary>
		''' Handles the OK button's Click event.
		''' </summary>
		''' <param name="sender">The button object.</param>
		''' <param name="e">The event arguments.</param>
		Private Sub btnOk_Click(ByVal sender As Object, ByVal e As EventArgs) Handles btnOk.Click
			' Check for invalid characters.
			If NewName.Contains("/") OrElse NewName.Contains("\") Then
				MessageBox.Show("Character '/' or '\' should not be entered", "New Name Prompt", MessageBoxButtons.OK, MessageBoxIcon.Stop)
				Return
			End If

			Me.DialogResult = System.Windows.Forms.DialogResult.OK
		End Sub
	End Class
End Namespace